### PR TITLE
Refactor Inspector resource group resource to use keyvaluetags package (or not)

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #10688.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSInspectorResourceGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSInspectorResourceGroup_basic -timeout 120m
=== RUN   TestAccAWSInspectorResourceGroup_basic
=== PAUSE TestAccAWSInspectorResourceGroup_basic
=== CONT  TestAccAWSInspectorResourceGroup_basic
--- PASS: TestAccAWSInspectorResourceGroup_basic (34.56s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	34.602s
```

It turns out that the `tags` attribute of the [`aws_inspector_resource_group`](https://www.terraform.io/docs/providers/aws/r/inspector_resource_group.html) resource doesn't tag the Inspector resource group but is used to select EC2 instances for inclusion in the resource group.

* Updated documentation to clarify this
* Removed `tagsInspector.go` and moved its functionality to `expand` (and `flatten`) functions in the same source file as the resource code
* Modified the resource Read function to set the `arn` and `tags` attributes
* Made the acceptance test a bit more complete

The [`aws_inspector_assessment_template`](https://www.terraform.io/docs/providers/aws/r/inspector_assessment_template.html) could support resource tags - https://github.com/terraform-providers/terraform-provider-aws/issues/11574 - so the generated code in the `keyvaluetags` package should stay.

### CODE REVIEW

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/5349

File: aws/resource_aws_inspector_resource_group_test.go [hunk 1]
```diff
@@ -17,40 +23,67 @@ func TestAccAWSInspectorResourceGroup_basic(t *testing.T) {
...
 				),
 			},
 			{
 				Config: testAccCheckAWSInspectorResourceGroupModified,
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckAWSInspectorResourceGroupExists("aws_inspector_resource_group.foo"),
+					testAccCheckAWSInspectorResourceGroupExists(resourceName, &v2),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "inspector", regexp.MustCompile(`resourcegroup/.+`)),
+					resource.TestCheckResourceAttr(resourceName, "tags.Name", "bar"),
+					func(s *terraform.State) error {
```
Comment (bflad) [inline aws/resource_aws_inspector_resource_group_test.go]: Nit: We typically create these as separate `TestCheckFunc` so they are reusable across multiple tests if desired, see also other `testAccCheck...Recreated` functions 👍

```
func testAccCheckAWSInspectorResourceGroupRecreated(i, j *inspector.ResourceGroup) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		if aws.TimeValue(i.CreatedAt) == aws.TimeValue(j.CreatedAt) {
			return fmt.Errorf("Inspector Resource Group not recreated")
		}
		return nil
	}
}
```

File: website/docs/r/inspector_resource_group.html.markdown [hunk 2]
```diff
@@ -3,17 +3,17 @@ subcategory: "Inspector"
...
 page_title: "AWS: aws_inspector_resource_group"
 description: |-
-  Provides a Inspector resource group.
+  Provides an Amazon Inspector resource group resource.
 ---
 
 # Resource: aws_inspector_resource_group
 
-Provides a Inspector resource group
+Provides a Amazon Inspector resource group resource.
```
Comment (bflad) [inline website/docs/r/inspector_resource_group.html.markdown]:
```suggestion
Provides an Amazon Inspector resource group resource.
```

File: website/docs/r/inspector_resource_group.html.markdown [hunk 3]
```diff
@@ -25,7 +25,7 @@ resource "aws_inspector_resource_group" "bar" {
 
 The following arguments are supported:
 
-* `tags` - (Required) The tags on your EC2 Instance.
+* `tags` - (Required) The tags tags (key and value pairs) that are used to select the EC2 instances to be included in an [Amazon Inspector assessment target](/docs/providers/aws/r/inspector_assessment_target.html).
```
Comment (bflad) [inline website/docs/r/inspector_resource_group.html.markdown]:
```suggestion
* `tags` - (Required) Key-value map of tags that are used to select the EC2 instances to be included in an [Amazon Inspector assessment target](/docs/providers/aws/r/inspector_assessment_target.html).
```

File: aws/resource_aws_inspector_resource_group_test.go (hunk 1 above)
Comment (ewbankkit) [reply aws/resource_aws_inspector_resource_group_test.go]: Done.

**Comment by ewbankkit** [comment]
@bflad Requested changes made.
Re-ran acceptance test:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSInspectorResourceGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSInspectorResourceGroup_basic -timeout 120m
=== RUN   TestAccAWSInspectorResourceGroup_basic
=== PAUSE TestAccAWSInspectorResourceGroup_basic
=== CONT  TestAccAWSInspectorResourceGroup_basic
--- PASS: TestAccAWSInspectorResourceGroup_basic (34.92s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	34.959s
```

**Pull request merged**